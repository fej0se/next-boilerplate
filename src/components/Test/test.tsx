import { render, screen } from '@testing-library/react'

import Test from '.'

describe('<Test />', () => {
  it('should render the heading', () => {
    const { container } = render(<Test />)

    expect(
      screen.getByRole('heading', { name: /This is a component test/i })
    ).toBeInTheDocument()
    expect(
      screen.getByRole('heading', { name: /Next.js boilerplate/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render color correctly', () => {
    const { container } = render(<Test />)

    expect(container.firstChild).toHaveStyle({ color: '#000000' })
  })

  it('should render background-color correctly', () => {
    const { container } = render(<Test />)

    expect(container.firstChild).toHaveStyle({ 'background-color': '#fefefe' })
  })
})
