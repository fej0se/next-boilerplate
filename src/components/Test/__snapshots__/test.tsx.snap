// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Test /> should render the heading 1`] = `
<main
  class="styles__wrapper-sc-unvff0-0 kimlBj"
>
  <h1
    class="styles__title-sc-unvff0-1 bCsivh"
  >
    Next.js boilerplate
  </h1>
  <h2
    class="styles__description-sc-unvff0-2 iKoZew"
  >
    This is a component test.
  </h2>
</main>
`
